### PR TITLE
fixed the issue where we tried to regex match against label=None

### PR DESCRIPTION
**Pull request by dx-pbuckley**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -168,7 +168,7 @@ def trigger():
                     issue_number = None
                     issue_comment = None
                     issue_url = None
-                    label = None
+                    label = 'github-webhooks-it-was-not-labeled'
```
Comment (ferrants) [inline main.py]: i don't think you want to change this from `None`, the label isn't `github-webhooks-it-was-not-labeled`

I think the issue was with the conditional checking the valid of label, try changing https://github.com/dataxu/github-webhooks/blob/master/main.py#L263 to
```
'label' not in matcher or label and re.search(matcher['label'], label)

see:
```
>>> matcher = {'label': 'int-ready'}
>>> label = "int-ready"
>>> ('label' not in matcher or re.search(matcher['label'], label) and label)
'int-ready'
>>> bool('label' not in matcher or re.search(matcher['label'], label) and label)
True
>>> label = None
>>> bool('label' not in matcher or re.search(matcher['label'], label) and label)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/mferrante/code/platform-integration/test_env2/lib/python2.7/re.py", line 142, in search
    return _compile(pattern, flags).search(string)
TypeError: expected string or buffer
>>> bool('label' not in matcher or label and re.search(matcher['label'], label))
False
```

```

File: main.py [hunk 2]
```diff
@@ -260,7 +260,7 @@ def trigger():
                                  ('repo' not in matcher or re.match(matcher['repo'], repo)) and
                                  ('message' not in matcher or re.search(matcher['message'], message)) and
                                  ('no_message' not in matcher or not re.search(matcher['no_message'], message)) and
-                                 ('label' not in matcher or re.search(matcher['label'], label) and label) and
+                                 ('label' not in matcher or label and re.search(matcher['label'], label)) and
```
Comment (dx-pbuckley) [inline main.py]: @ferrants looks better?

Comment (ferrants) [inline main.py]: :+1:

**Comment by ferrants** [comment]
:shipit: if tests pass

**Pull request merged**